docs/api-conventions.md

```diff
@@ -127,6 +164,8 @@ Kubernetes by convention exposes additional verbs as new endpoints with singular
 
 Support of additional verbs is not required for all object types.
 
+When resources wish to expose alternative actions, they should do so using new sub-resources or as new top level resources. An example is allowing automated processes to update the "status" field of a Pod. The `/pods` endpoint only allows updates to "metadata" and "spec", since those reflect user intent. An automated process should be able to modify status for users to see by sending an updated Pod kind to the server to the "/pods/&lt;name&gt;/status" endpoint - the alternate endpoint allows different rules to be applied to the update, and access to be appropriately restricted.
```

Comment:
I feel we need some guidance about when each might be appropriate.

IMO, /proxy and /redirect aren't really API calls. They are apiserver services.

/watch is sort of like a custom HTTP verb. It doesn't change the resources returned. 

What all these prefix-based paths have in common is that they are transport-oriented.

I prefer the more object-oriented subresource approach for the cases discussed in this PR: binding, status, size, stop, restart, clone, suspend, etc.